lib/mirage/impl/mirage_impl_block.ml

```diff
@@ -188,19 +189,20 @@ let docteur_unix (mode : mode) extra_deps disk branch analyze remote =
       {ocaml|let ( <.> ) f g = fun x -> f (g x) in
              let f = Rresult.R.(failwith_error_msg <.> reword_error (msgf "%%a" %s.pp_error)) in
              Lwt.map f (%s.connect ~analyze:%a %a)|ocaml}
-      modname modname Key.serialize_call (Key.v analyze) Key.serialize_call
-      (Key.v disk)
+      modname modname Runtime_key.call analyze Runtime_key.call disk_r
```

Comment:
/cc @dinosaure: I am not sure what docteur does here - the disk key seems to be used at runtime to pick the name of the disk? It might makes sense on unix, but what's the semantics on solo5?

Also: I've added 2 disk parameters (one for configure-time, one for runtime) but now only the configure-time one is tweakable by the users for now. It's a good idea to separate those anyway as the location of the disk is unlikely to be similar on the machine that builds the unikernel and the machine that deploys it.

What do you think? Are you ok with two different arguments here? And what should be the default value on solo5?

Reply:
I will try to check what happens next week (please, give me the time to see if we should rethink about how we handle block device, I was not very satisfied by what I did before).

Reply:
no hurry - that's a big change; let me know when you are happy with it.

TBH, I'm not 100% sure about exposing the value of the configure-time key to the runtime (as a read-only value) or not. We do not do this anymore with this patch, but it wouldn't be too hard to re-add this, and maybe that'll make the life of `docteur` easier.